### PR TITLE
Add rpy2@3.0.4

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-rpy2/package.py [hunk 1]
```diff
@@ -30,11 +31,15 @@ class PyRpy2(PythonPackage):
...
+    # @3.0.0:
+    depends_on('py-cffi@1.0.0:', when='@3.0.0:', type=('build', 'run'))
+    depends_on('py-simplegeneric', when='@3.0.0:', type=('build', 'run'))
+
     # @2.9.0:
     depends_on('r@3.3:',    when='@2.9.0:', type=('build', 'run'))
     depends_on('python@3:', when='@2.9.0:', type=('build', 'run'))
     depends_on('py-jinja2', when='@2.9.0:', type=('build', 'run'))
-    depends_on('py-six',    when='@2.9.0:', type=('build', 'run'))
+    depends_on('py-six',    when='@2.9.0:2.9.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rpy2/package.py]: Can you add:
```python
depends_on('python@3.5:', when='@2.9:', type=('build', 'run'))
depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
depends_on('py-pytest', when='@3:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-rpy2/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-rpy2/package.py]: I'm not sure about Python version restriction: [the commit](https://github.com/rpy2/rpy2/commit/03c1a76a02730337e2e45f21af20cd0aecb748a0) was done before 3.1.0 release, but the wording may imply that `Python >= 3.5` (and not just `>= 3`) was required from 2.9.0 onward

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-rpy2/package.py]: Yeah, I think they always required 3.5 but never specified. At the very least, Python 2 won’t work starting with 2.9

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-rpy2/package.py]: Yup: https://github.com/rpy2/rpy2/blob/RELEASE_2_9_0/setup.py#L411.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**